apps/wallet/src/ui/app/helpers/getMoveTxnSummary.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { getMoveCallTransaction } from '@mysten/sui.js';
+
+import type { SuiTransactionKind } from '@mysten/sui.js';
+
+const stakingCalls = [
+    'request_add_delegation',
+    'request_add_stake_with_locked_coin',
+    'request_withdraw_stake',
+    'request_add_delegation_with_locked_coin',
+    'request_withdraw_delegation',
+    'request_switch_delegation',
+];
+
+// Get known native move function
+export function getMoveCallMeta(txDetails: SuiTransactionKind): {
+    label: string;
+    fnName: string;
+    validatorAddress?: string | null;
+} | null {
+    const moveCall = getMoveCallTransaction(txDetails);
+    if (!moveCall) return null;
+
+    let label = 'Move Call';
+    let validatorAddress;
+    const fnName = moveCall.function.replace(/_/g, ' ');
+
+    if (moveCall.module === 'devnet_nft' && moveCall.function === 'mint') {
```

Comment:
I don't like this custom handling, I'd remove it to better set expectations with how these will log.